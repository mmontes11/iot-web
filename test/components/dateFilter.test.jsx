import React from "react";
import DateFilter from "components/dateFilter";
import IntlProvider from "containers/intlProvider";
import store from "config/store";
import { shallow } from "enzyme";

describe("components/dateFilter", () => {
  it("renders DateFilter for selecting a time period", () => {
    const dateFilter = {
      selector: {
        isCustomSelected: false,
        onChange: () => undefined,
      },
      timePeriod: {
        label: "Select time period:",
        isLoading: false,
        isActive: true,
        isDisabled: false,
        items: ["foo", "bar"],
        selectedItem: null,
        onButtonClick: () => undefined,
        onItemClick: () => undefined,
      },
      custom: {
        startDate: {
          selected: null,
          onChange: () => undefined,
        },
        endDate: {
          selected: null,
          onChange: () => undefined,
        },
      },
      onDelete: () => undefined,
    };
    const wrapper = shallow(
      <IntlProvider store={store}>
        <DateFilter dateFilter={dateFilter} />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders DateFilter for selecting a custom time period", () => {
    const dateFilter = {
      selector: {
        isCustomSelected: true,
        onChange: () => undefined,
      },
      timePeriod: {
        isLoading: false,
        isActive: true,
        isDisabled: false,
        items: ["foo", "bar"],
        selectedItem: null,
        onButtonClick: () => undefined,
        onItemClick: () => undefined,
      },
      custom: {
        startDate: {
          selected: null,
          onChange: () => undefined,
        },
        endDate: {
          selected: null,
          onChange: () => undefined,
        },
      },
      onDelete: () => undefined,
    };
    const wrapper = shallow(
      <IntlProvider store={store}>
        <DateFilter dateFilter={dateFilter} />
      </IntlProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("simulates clicks on buttons", () => {
    const onSelectorChange = jest.fn();
    const onDelete = jest.fn();
    const dateFilter = {
      selector: {
        isCustomSelected: true,
        onChange: onSelectorChange,
      },
      timePeriod: {
        isLoading: false,
        isActive: true,
        isDisabled: false,
        items: ["foo", "bar"],
        selectedItem: null,
        onButtonClick: () => undefined,
        onItemClick: () => undefined,
      },
      custom: {
        startDate: {
          selected: null,
          onChange: () => undefined,
        },
        endDate: {
          selected: null,
          onChange: () => undefined,
        },
      },
      onDelete,
    };
    const wrapper = shallow(
      <IntlProvider store={store}>
        <DateFilter dateFilter={dateFilter} />
      </IntlProvider>,
    );
    wrapper.find("#time-period-button").simulate("click");
    expect(onSelectorChange).toHaveBeenCalled();
    wrapper.find("#delete-button").simulate("click");
    expect(onDelete).toHaveBeenCalled();
  });
});
