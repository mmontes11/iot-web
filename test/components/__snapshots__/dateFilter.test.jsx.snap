// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/dateFilter renders DateFilter for selecting a custom time period 1`] = `
<div
  className="box date-filter"
>
  <div>
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <DatePicker
          onChange={[Function]}
          placeholder="Select start date:"
          selected={null}
        />
      </div>
    </div>
    <div
      className="columns"
    >
      <div
        className="column"
      >
        <DatePicker
          onChange={[Function]}
          placeholder="Select end date:"
          selected={null}
        />
      </div>
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="column is-10"
    >
      <button
        className="button is-info"
        id="time-period-button"
        onClick={[Function]}
      >
        Time Period
      </button>
    </div>
    <div
      className="column is-center"
    >
      <button
        className="delete is-medium"
        id="delete-button"
        onClick={[Function]}
        tabIndex={0}
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;

exports[`components/dateFilter renders DateFilter for selecting a time period 1`] = `
<div
  className="box date-filter"
>
  <div
    className="columns"
  >
    <div
      className="column"
    >
      <Dropdown
        buttonStyle=""
        iconStyle="fa-angle-down"
        isActive={true}
        isDisabled={false}
        isLoading={false}
        items={
          Array [
            "foo",
            "bar",
          ]
        }
        label="Select time period:"
        onButtonClick={[Function]}
        onItemClick={[Function]}
        selectedItem={null}
      />
    </div>
  </div>
  <div
    className="columns"
  >
    <div
      className="column is-10"
    >
      <button
        className="button is-info"
        id="time-period-button"
        onClick={[Function]}
      >
        Custom
      </button>
    </div>
    <div
      className="column is-center"
    >
      <button
        className="delete is-medium"
        id="delete-button"
        onClick={[Function]}
        tabIndex={0}
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
